package android.appwidget;

public class AppWidgetProviderInfo implements android.os.Parcelable {
    public static final android.os.Parcelable.Creator<android.appwidget.AppWidgetProviderInfo> CREATOR = null;
    public static final int RESIZE_BOTH = 3;
    public static final int RESIZE_HORIZONTAL = 1;
    public static final int RESIZE_NONE = 0;
    public static final int RESIZE_VERTICAL = 2;
    public static final int WIDGET_CATEGORY_HOME_SCREEN = 1;
    public static final int WIDGET_CATEGORY_KEYGUARD = 2;
    public static final int WIDGET_CATEGORY_NOT_KEYGUARD = 8;
    public static final int WIDGET_CATEGORY_SEARCHBOX = 4;
    public static final int WIDGET_CATEGORY_UNKNOWN = -1;
    public static final int WIDGET_FEATURE_CONFIGURATION_OPTIONAL = 4;
    public static final int WIDGET_FEATURE_HIDE_FROM_PICKER = 2;
    public static final int WIDGET_FEATURE_RECONFIGURABLE = 1;
    public int autoAdvanceViewId;
    public android.content.ComponentName configure;
    public int descriptionRes;
    public int generatedPreviewCategories;
    public int icon;
    public int initialKeyguardLayout;
    public int initialLayout;
    public boolean isExtendedFromAppWidgetProvider;
    @java.lang.Deprecated
    public java.lang.String label;
    public int maxResizeHeight;
    public int maxResizeWidth;
    public int minHeight;
    public int minResizeHeight;
    public int minResizeWidth;
    public int minWidth;
    public int previewImage;
    public int previewLayout;
    public android.content.ComponentName provider;
    public android.content.pm.ActivityInfo providerInfo;
    public int resizeMode;
    public int targetCellHeight;
    public int targetCellWidth;
    public int updatePeriodMillis;
    public int widgetCategory;
    public int widgetFeatures;
    public AppWidgetProviderInfo() {}
    public AppWidgetProviderInfo(android.os.Parcel p0) {}
    private android.graphics.drawable.Drawable loadDrawable(android.content.Context p0, int p1, int p2, boolean p3) { return null; }
    public android.appwidget.AppWidgetProviderInfo clone() { return null; }
    public int describeContents() { return 0; }
    public android.content.pm.ActivityInfo getActivityInfo() { return null; }
    public final android.os.UserHandle getProfile() { return null; }
    public final java.lang.CharSequence loadDescription(android.content.Context p0) { return null; }
    public final android.graphics.drawable.Drawable loadIcon(android.content.Context p0, int p1) { return null; }
    public final java.lang.String loadLabel(android.content.pm.PackageManager p0) { return null; }
    public final android.graphics.drawable.Drawable loadPreviewImage(android.content.Context p0, int p1) { return null; }
    public java.lang.String toString() { return null; }
    public void updateDimensions(android.util.DisplayMetrics p0) {}
    public void writeToParcel(android.os.Parcel p0, int p1) {}

    @java.lang.annotation.Retention(java.lang.annotation.RetentionPolicy.SOURCE)
    public static @interface CategoryFlags {
    }

    @java.lang.annotation.Retention(java.lang.annotation.RetentionPolicy.SOURCE)
    public static @interface FeatureFlags {
    }

    @java.lang.annotation.Retention(java.lang.annotation.RetentionPolicy.SOURCE)
    public static @interface ResizeModeFlags {
    }
}
